Examples/Io/Csv/src/CsvBFieldWriter.cpp

```diff
@@ -0,0 +1,219 @@
+          }
+        }
+      }
+    }
+  } else {
+    ACTS_INFO("Writing RZ field of size " << (bins[0] + 1) << " x "
+                                          << (bins[1] + 1) << " to file "
+                                          << config.fileName << "...");
+
+    std::size_t total_items = (bins[0] + 1) * (bins[1] + 1);
+
+    // For cylindrical coordinates, we only need to iterate over the r and z
+    // coordinates, because we assume rotational cylindrical symmetry. This
+    // makes the procedure quite a bit faster, too. Great!
+    for (std::size_t r = 0; r <= bins[0]; ++r) {
+      for (std::size_t z = 0; z <= bins[1]; ++z) {
+        // Calculate the position (still in three dimensions), assuming that
+        // the phi coordinate is zero. Then grab the field.
+        Acts::Vector3 pos(min[0] + r * delta[0], 0.f, min[1] + z * delta[1]);
+
+        Acts::Result<Acts::Vector3> flx = field.getField(pos, cache);
+
+        // Check the result, then write to disk. We write the r and z positions
+        // as they are, then we write the z component of the result vector as
+        // is, and we compute the r-value from the other components of the
+        // vector.
+        if (flx.ok()) {
+          writer.append(
+              pos[0] / Acts::UnitConstants::mm,
+              pos[2] / Acts::UnitConstants::mm,
+              Acts::VectorHelpers::perp(*flx) / Acts::UnitConstants::T,
+              (*flx)[2] / Acts::UnitConstants::T);
+        } else {
+          throw std::runtime_error("B-field returned a non-extant value!");
+        }
+
+        // As before, print some progress reports for the user to enjoy while
+        // they wait.
+        std::size_t idx = (r * (bins[1] + 1)) + z + 1;
+
+        if (idx % 10000 == 0 || idx == total_items) {
+          ACTS_VERBOSE("Wrote " << idx << " out of " << total_items
+                                << " items (" << std::setprecision(3)
+                                << ((100.f * idx) / total_items) << "%).");
+        }
+      }
+    }
+  }
+}
+
+// Note that this is a C++ source file, and not a header file. The reason for
+// this is that we can easily enumerate the different template parameter
+// combinations for the function above, and so we want to speed up compilation
+// times by just compiling these symbols once into the object file
+// corresponding to this TU.
+template void CsvBFieldWriter::run<CsvBFieldWriter::CoordinateType::XYZ, true>(
+    const Config<CoordinateType::XYZ, true>&, Acts::Logging::Level);
+template void CsvBFieldWriter::run<CsvBFieldWriter::CoordinateType::RZ, true>(
+    const Config<CoordinateType::RZ, true>&, Acts::Logging::Level);
+template void CsvBFieldWriter::run<CsvBFieldWriter::CoordinateType::XYZ, false>(
+    const Config<CoordinateType::XYZ, false>&, Acts::Logging::Level);
+template void CsvBFieldWriter::run<CsvBFieldWriter::CoordinateType::RZ, false>(
+    const Config<CoordinateType::RZ, false>&, Acts::Logging::Level);
```

Comment:
Doesn't hurt but did you measure if this has an effect? I believe the compiler will instantiate the template in other TU anyway, and just discard duplicates during linking (like with `inline`).

Reply:
Right, but that still increases compile times in those other TUs, right? I haven't tested this at all, but it seemed to me like hiding them implementation from other translation units entirely would be the most sound way of doing this. But I am open to change it.

Reply:
I think what I'm saying is I don't think this does anything.

Reply:
Well in the end I don't feel strongly either way; if you prefer to not have the specializations here I can move the code to a header file and it should work. Let me know what you prefer.